torch_geometric/nn/summary.py

```diff
@@ -0,0 +1,146 @@
+from collections import defaultdict
+from typing import Any, Optional, Sequence, Tuple, Union
+
+import torch
+import torch.nn as nn
+from torch.jit import ScriptModule
+from torch_sparse import SparseTensor
+
+from torch_geometric.nn.conv import MessagePassing
+
+
+def summary(
+    model: nn.Module,
+    *inputs: Sequence[Any],
+    max_depth: int = 3,
+    leaf_module: Union[nn.Module, Tuple[nn.Module],
```

Comment:
```suggestion
    leaf_modules: List[Union[nn.Module, Tuple[nn.Module],
```
Should  this be a list? Maybe multiple modules you want to treat this way?

Reply:
No, it is used to check if the input model `isinstance(model, A)` or `isinstance(model, (A, B))`. So it's better to be a tuple if multiple modules are provided.

Reply:
Oh I see your point 👍